### PR TITLE
feat(toc-sidebar): Add Collapsible Lists

### PR DESCRIPTION
**Pull request by david-clark-1043**
The Table of Contents (TOC) was unwieldy. Added bootstrap collapse classes/functionality to list elements that have a sub-list element. Clicking the [+] beside these elements expands the collapsed sub-list. JS should replace the [+] with a [–]. Clicking the [–] should collapse the sub-list. JS should replace the [–] with a [+].

Part of: #2029
  
Collapsed  
  
toc_collapsed  
  
Expanded  
  
toc_expanded

### CODE REVIEW

**Comment by mlissner** [comment]
Looks good. One thought: Bootstrap has a collapse feature. Did you look at that: https://getbootstrap.com/docs/4.3/components/collapse/

(I don't know what semgrep is angry about, but we can ignore it.)

**Comment by david-clark-1043** [comment]
> Looks good. One thought: Bootstrap has a collapse feature. Did you look at that: https://getbootstrap.com/docs/4.3/components/collapse/
> 
> (I don't know what semgrep is angry about, but we can ignore it.)

Yep! That's what I used. I only had to add a little bit of custom javascript in base.js to change the icon visually which I thought was useful for the user. It has a check in the js that triggers because if the collapsible elements are nested, the expanding/collapsing the inner one would trigger the icon change for all of the outer elements. A pretty straightforward if statement took care of that though to make sure the elements are paired correctly.

**Comment by mlissner** [comment]
Oh shoot, you even mentioned the boostrap thing in your comment. Sorry.

File: cl/assets/static-global/js/base.js [hunk 1]
```diff
@@ -176,6 +176,28 @@ $(document).ready(function () {
     }
   });
 
+  ///////////////////////////
+  // TOC Collapse Controls //
+  ///////////////////////////
+
+  $('.collapse').on('shown.bs.collapse', function () {
+    // do something…
```
Comment (mlissner) [inline cl/assets/static-global/js/base.js:184]: Can you improve these comments, please, and then I think we're good to go.

**Review by mlissner — CHANGES_REQUESTED** [review]
Thanks. Just one quick thing.

File: cl/assets/static-global/js/base.js (hunk 1 above)
Comment (david-clark-1043) [reply cl/assets/static-global/js/base.js:184]: Oh, yes, sorry! That's the placeholder from the bootstrap example I copied.

**Comment by mlissner** [comment]
Great. Thank you! I look forward to the next pieces!

**Pull request merged**